cachito/workers/pkg_managers/pip.py

```diff
@@ -635,3 +637,347 @@ def _get_variable(self, var_name):
+    LINE_COMMENT = re.compile(r"(^|\s)+#.*$")
+
+    # Options allowed in a requirements file. The values represent whether or not the option
+    # requires a value.
+    # https://pip.pypa.io/en/stable/reference/pip_install/#requirements-file-format
+    OPTIONS = {
+        "--constraint": True,
+        "--editable": False,  # The required value is the requirement itself, not a parameter
+        "--extra-index-url": True,
+        "--find-links": True,
+        "--index-url": True,
+        "--no-binary": True,
+        "--no-index": False,
+        "--only-binary": True,
+        "--pre": False,
+        "--prefer-binary": False,
+        "--require-hashes": False,
+        "--requirement": True,
+        "--trusted-host": True,
+        "--use-feature": True,
+        "-c": True,
+        "-e": False,  # The required value is the requirement itself, not a parameter
+        "-f": True,
+        "--hash": True,
+        "-i": True,
+        "-r": True,
+    }
+
+    # Options that are specific to a single requirement in the requirements file. All other
+    # options apply to all the requirements.
+    REQUIREMENT_OPTIONS = {"-e", "--editable", "--hash"}
+
+    def __init__(self, file_path):
+        """Initialize a PipRequirementsFile.
+
+        :param str file_path: the full path to the requirements file
+        """
+        self.file_path = file_path
+        self.__parsed = NOTHING
+
+    @property
+    def requirements(self):
+        """Return a list of PipRequirement objects."""
+        return self._parsed["requirements"]
+
+    @property
+    def options(self):
+        """Return a list of options."""
+        return self._parsed["options"]
+
+    @property
+    def _parsed(self):
+        """Return the parsed requirements file.
+
+        :return: a dict with the keys ``requirements`` and ``options``
+        """
+        if self.__parsed is NOTHING:
+            parsed = {"requirements": [], "options": []}
+
+            for line in self._read_lines():
+                (
+                    global_options,
+                    requirement_options,
+                    requirement_line,
+                ) = self._split_options_and_requirement(line)
+                if global_options:
+                    parsed["options"].extend(global_options)
+
+                if requirement_line:
+                    parsed["requirements"].append(
+                        PipRequirement.from_line(requirement_line, requirement_options)
+                    )
+
+            self.__parsed = parsed
+
+        return self.__parsed
+
+    def _read_lines(self):
+        """Read and yield the lines from the requirements file.
+
+        Lines ending in the line continuation character are joined with the next line.
+        Comment lines are ignored.
+        """
+        buffered_line = []
+
+        with open(self.file_path) as f:
+            for line in f.read().splitlines():
+                if self.LINE_COMMENT.match(line):
```

Comment:
This should be `search` instead of `match` to capture comments in the end of a non-comment line. It would be nice to add a unit test for such case as well.

Reply:
We don't want to ignore such lines though, right? We could do something like this instead:
```python
comment =  re.compile(r"\s*#.*")
line = comment.sub("", line)
if not line:
    continue
```

Reply:
+1

Note that `re.compile(r"\s*#.*")` may introduce an issue. it would remove `#123` from `foo#123`.  Maybe we could go with `re.compile(r"(^|\s)#.*")`.

The additional spaces before the last one preceding the `#` would need to be stripped later (alternativelly, we could use the - potentially slower - `+` quantifier).

Reply:
Thank you for catching this. I was not handling inline comments properly, e.g. `spam==1.2.3 # ignore me`.

Reply:
This is now handled, let me know what you think.